### PR TITLE
coverage improvement for istio.io/istio/pilot/pkg/model

### PR DESCRIPTION
**Pull request by stewart-yu**
xref #8568
before change:
```
istio.io/istio/pilot/pkg/model/authorization.go:51:	addServiceRole				0.0%
istio.io/istio/pilot/pkg/model/authorization.go:68:	addServiceRoleBinding			0.0%
istio.io/istio/pilot/pkg/model/authorization.go:96:	AddConfig				0.0%
istio.io/istio/pilot/pkg/model/authorization.go:112:	RolesForNamespace			0.0%
istio.io/istio/pilot/pkg/model/authorization.go:126:	RoleToBindingsForNamespace		0.0%
istio.io/istio/pilot/pkg/model/authorization.go:140:	newAuthzPolicies			0.0%
...
istio.io/istio/pilot/pkg/model/config.go:565:		ServiceEntries				0.0%
istio.io/istio/pilot/pkg/model/config.go:574:		sortConfigByCreationTime		0.0%
istio.io/istio/pilot/pkg/model/config.go:604:		EnvoyFilter				0.0%
istio.io/istio/pilot/pkg/model/config.go:635:		HTTPAPISpecByDestination		0.0%
```
and coverage of this package is 
```
coverage: 66.70% of statements
```
But after change
```
istio.io/istio/pilot/pkg/model/authorization.go:51:	addServiceRole				87.5%
istio.io/istio/pilot/pkg/model/authorization.go:68:	addServiceRoleBinding			92.9%
istio.io/istio/pilot/pkg/model/authorization.go:96:	AddConfig				83.3%
istio.io/istio/pilot/pkg/model/authorization.go:112:	RolesForNamespace			100.0%
istio.io/istio/pilot/pkg/model/authorization.go:126:	RoleToBindingsForNamespace		100.0%
....
istio.io/istio/pilot/pkg/model/config.go:565:		ServiceEntries				75.0%
istio.io/istio/pilot/pkg/model/config.go:574:		sortConfigByCreationTime		66.7%
istio.io/istio/pilot/pkg/model/config.go:604:		EnvoyFilter				71.4%
istio.io/istio/pilot/pkg/model/config.go:635:		HTTPAPISpecByDestination		91.7%
```
and coverage of this package is 
```
coverage: 71.3% of statements
```
greater than `70%` defined in [https://github.com/istio/istio/blob/master/codecov.requirement](https://github.com/istio/istio/blob/master/codecov.requirement)

### CODE REVIEW

File: pilot/pkg/model/validation_test.go [hunk 1]
```diff
@@ -3119,3 +3119,50 @@ func TestValidateRbacConfig(t *testing.T) {
...
+			name: "invalid domian or labels",
+			in:   &mccpb.IstioService{Name: "test-service-name", Domain: strings.Repeat("x", 256)},
+		},
+		{
+			name:  "valid",
+			in:    validService,
+			valid: true,
+		},
+	}
+	for _, c := range cases {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/validation_test.go]: use subtest https://blog.golang.org/subtests

Comment (stewart-yu) [reply pilot/pkg/model/validation_test.go]: fixed

**Pull request merged**